osu.Game.Rulesets.Mania/UI/Components/ColumnType.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Mania.UI.Components
+{
+    public abstract class ColumnType
```

Comment:
`ColumnType` is a bad name for this. `ColumnInfo` maybe?

Also please add full xmldoc for the class and all properties.